python/ctsm/toolchain/gen_mksurfdata_namelist.py

```diff
@@ -491,271 +882,45 @@ def main():
             if item.tag == "lookup_filename":
                 new_key = f"{child1.tag}_lookup"
                 rawdata_files[new_key] = os.path.join(input_path, item.text)
+    return tool_path, must_run_download_input_data, rawdata_files
 
-    # determine output mesh
-    xml_path = os.path.join(tool_path, "../../ccs_config/component_grids_nuopc.xml")
-    tree2 = ET.parse(xml_path)
-    root = tree2.getroot()
-    model_mesh = ""
-    for child1 in root:  # this is domain tag
-        for _, value in child1.attrib.items():
-            if value == res:
-                for child2 in child1:
-                    if child2.tag == "mesh":
-                        model_mesh = child2.text
-                        rawdata_files["mksrf_fgrid_mesh"] = os.path.join(
-                            input_path, model_mesh.strip("$DIN_LOC_ROOT/")
-                        )
-                    if child2.tag == "nx":
-                        rawdata_files["mksrf_fgrid_mesh_nx"] = child2.text
-                    if child2.tag == "ny":
-                        rawdata_files["mksrf_fgrid_mesh_ny"] = child2.text
 
-    if not model_mesh and force_model_mesh_file is None:
-        valid_grids = []
-        for child1 in root:  # this is domain tag
-            for _, value in child1.attrib.items():
-                valid_grids.append(value)
-        if res in valid_grids:
+def open_mesh_file(force_model_mesh_file, force_model_mesh_nx, force_model_mesh_ny):
+    """
+    open mesh_file to read element_count and, if available, orig_grid_dims
+    """
+    # pylint: disable=no-name-in-module,no-member
```

Comment:
I'm not sure what it's complaining about here. I can look it up, but if you can easily explain that would be great.

Reply:
I can't explain it, honestly. It thinks the `netCDF4` module doesn't have a member called `Dataset`, but when I open a Python console in `ctsm_pylib` and do `from netCDF4 import Dataset` it works fine.

Reply:
OK, just pylint strangeness. It might be good to mark with a comment that says above. We likely will need to use this in other places...

Reply:
Resolved with d8b6a5d80.